File: problems/[E]326.Power-of-Three.kt
/**
 * Runtime: 228 ms, faster than 82.93% of Kotlin online submissions for Power of Three.
 * Memory Usage: 35.3 MB, less than 92.68% of Kotlin online submissions for Power of Three.
 */
class Solution {
    fun isPowerOfThree(n: Int): Boolean {
        if (n == 0) return false
        
        var number = n
        while (number % 3 == 0) {
            number /= 3
        }
        return number == 1
    }
}

File: python/[E]326.Power-of-Three.py
# Runtime: 56 ms, faster than 98.99% of Python3 online submissions for Power of Three.
# Memory Usage: 14 MB, less than 92.09% of Python3 online submissions for Power of Three.
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n <= 0: return False
        while n % 3 == 0:
            n /= 3
        return n == 1
        
